patches/RoboCon2024_p2/home/pi/robot/robot/vision.py

```diff
@@ -276,17 +277,26 @@ def res(self):
     @res.setter
     def res(self, new_res):
         if new_res is not self._res:
+            '''cv_property_ids = (cv2.CAP_PROP_FRAME_WIDTH,
```

Comment:
Just wondering what this is? Is it debug code which is commented out?

Just wondering if we need it in production?

Reply:
Thank you for handling this! 😊 .

I was poking around earlier and tried to weed out some of the random debug code from our commits. It's super important to keep that stuff from sneaking into production. It tends to clutter up the production code, and before you know it, you're left wondering what's what, why it's there, and what's happening. This clutter can make the codebase pretty unwieldy, to the point where adding new features becomes...... difficult to say the least.

This patch and bug you're tackling is a classic example – it’s fixing some commented out code that somehow made its way to production.

We need to make sure that commented out code doesn't do this again!

Reply:
Also as a side note, using strings as multi-line comments is problematic becuase it can cause unexpected behaviour as a string is an actual object which you are creating. Its also non-standard style: https://peps.python.org/pep-0008/#block-comments 

Your editor should be able to prepend a `#` instead to the start of each line. For VSCode this is `CTRL + /`, I think.

Its okay to use strings as comments for functions as python is then able to lift that into the language. The comment then becomes part of the code. e.g.

```py
def my_function(foo, bar):
    """This function does magic
        foo - the first part to do magic on
        bar - the second part to do magic on
    """
    return foo + bar
    
print(my_function.__doc__)
```

The string here actually means something and you can interact with it from the langage. But don't randomly declare strings in the middle of the code.

Reply:
But for debug like you were using it this is completely fine!